migtests/scripts/resumption.py

```diff
@@ -0,0 +1,260 @@
+
+
+def run_and_resume_voyager(command, resumption):
+    """
+    Runs the yb-voyager command with support for resumption testing.
+    """
+    for attempt in range(1, resumption['max_restarts'] + 1):
+        print(f"\n--- Attempt {attempt} of {resumption['max_restarts']} ---")
+        try:
+            process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+            print("Running command:", ' '.join(command), flush=True)
+
+            start_time = time.time()
+            full_output = ''
+
+            while True:
+                rlist, _, _ = select.select([process.stdout, process.stderr], [], [], 5)
+                for ready in rlist:
+                    output = ready.readline()
+                    if not output:  # Exit if output is empty (end of process output)
+                        break
+                    full_output += output
+                if time.time() - start_time > 5:
+                    break
+
+            if full_output:
+                print(full_output.strip(), flush=True)
+
+            while True:
+                if process.poll() is not None:
+                    break  # Process has ended, exit loop
+
+                interrupt_interval_seconds = random.randint(
+                    resumption['min_interrupt_seconds'], 
+                    resumption['max_interrupt_seconds']
+                )
+                print(f"\nProcess will be interrupted in {interrupt_interval_seconds // 60}m {interrupt_interval_seconds % 60}s")
+                time.sleep(interrupt_interval_seconds)
+                print(f"\nInterrupting the import process (PID: {process.pid})")
+                process.send_signal(signal.SIGINT)
+
+                restart_wait_time_seconds = random.randint(
+                    resumption['min_restart_wait_seconds'], 
+                    resumption['max_restart_wait_seconds']
+                )
+                print(f"\nWaiting for {restart_wait_time_seconds // 60}m {restart_wait_time_seconds % 60}s before resuming...")
+                time.sleep(restart_wait_time_seconds)
+
+        except Exception as e:
+            print(f"Error occurred during import: {e}")
+            if process:
+                process.kill()
+            raise e
+        
+        finally:
+            if process and process.poll() is None:
+                print(f"Terminating process (PID: {process.pid})")
+                process.kill()
+                process.wait(timeout=30)
+
+    # Final import retry logic
+    print("\n--- Final attempt to complete the import ---")
+    
+    for _ in range(2): 
+        try:
+            print("\nVoyager command output:")
+
+            process = subprocess.Popen(
+                command, 
+                stdout=subprocess.PIPE, 
+                stderr=subprocess.PIPE, 
+                text=True
+            )
+
+            # Capture and print output
+            for line in iter(process.stdout.readline, ''):
```

Comment:
in the above for-loop, we're reading both stderr and stdout, here we're only reading stdout. Any particular reason? Would be good to be consistent here (call a common function that captures stdout/stderr)

Reply:
Also till when will you keep reading? How long will the loop run?

Reply:
The loop was to be run till the command exits and print the output in realtime.
Implemented a separate function for running the command and capturing stdout and stderr.